src/csvcubed/writers/qbwriter.py

```diff
@@ -811,6 +955,24 @@ def _generate_csvqb_column(self, column: CsvColumn) -> Dict[str, Any]:
 
         return csvw_col
 
+    def _get_observation_value_col_for_column(
+        self, col_title: str
+    ) -> QbColumn[QbObservationValue]:
+        """
+        Gets the matching observation value column for the given attributes/units column (if there is one).
+        """
+        obs_value_columns = get_columns_of_dsd_type(self.cube, QbObservationValue)
+        obs_columns_for_column = [
+            obs_col
+            for obs_col in obs_value_columns
+            if col_title == obs_col.csv_column_title
+        ]
+        if len(obs_columns_for_column) != 1:
+            raise Exception(
```

Comment:
Can we do a [KeyError](https://docs.python.org/3/library/exceptions.html#KeyError) here?